### DISCUSSION TITLE
v1.96.5

### DISCUSSION THREAD

**Discussion by cyring** [cyring/CoreFreq#453] [Show and tell]
### [AMD]

- Adding the "AMD Ryzen 5 5600X3D"
- Introducing the Zen4c Bergamo architecture
- Specs fixing of Embedded V-series and 3000 Series Processors
- Adding MENDOCINO, RAPHAEL Pro, PHOENIX Pro among others

<hr /><em>This discussion was created from the release <a href='https://github.com/cyring/CoreFreq/releases/tag/1.96.5'>v1.96.5</a>.</em>

**Comment by cyring** [comment]
* Live [ISO](https://github.com/cyring/CoreFreq/wiki/Live-CD) also refreshed

**Comment by huajian628** [comment]
why not have dkms.deb?

**Comment by huajian628** [comment]
Compiling from source is difficult

**Comment by huajian628** [comment]
Can someone create a DKMS.deb package that can be installed on non-Ubuntu distributions?

**Reply by cyring** [reply]
Hello, 
What do you find difficult in the installation following the dkms instructions and prerequisites listed into the Readme.md ?

**Reply by huajian628** [reply]
Hello, I have many computers that need to install this tool. What I prefer is to have a deb package or rpm package that can be installed directly on different system versions.

**Reply by cyring** [reply]
Thanks to their respective contributors, _CoreFreq_ has packages for:
* [openSuse](https://software.opensuse.org/package/CoreFreq) and [pkgs.org](https://pkgs.org/download/CoreFreq) 
* [Unraid](https://github.com/ich777/unraid-corefreq) 
* [Gentoo](https://github.com/gentoo/guru/tree/master/sys-apps/corefreq) 
* [ArchLinux](https://aur.archlinux.org/packages/CoreFreq-dkms) 
* [Guix](https://packages.guix.gnu.org/packages/corefreq)

**Reply by huajian628** [reply]
why not have ubuntu or centos?

**Reply by cyring** [reply]
Because no one came to take this in charge.

**Reply by huajian628** [reply]
I really urgently need the Ubuntu and CentOS distribution installation packages corresponding to this tool.

**Reply by cyring** [reply]
@huajian628
> that can be installed directly on different system versions.

The `corefreqk.ko` driver at the heart of _CoreFreq_ has to be built against the kernel version of each different target.

Because all your systems differs, you may need as much _CoreFreq_ binary package versions. 
A source-deb package, which will still need the kernel headers, will face same constraints.

What about building _CoreFreq_ for each target on development virtual machine ? 

A minimal VM based on each target completed with the development prerequisites to run `make`

Then push (`scp` or `rsync`) the 3 binaries `corefreqk.ko corefreqd corefreq-cli` from the VM to its associated production system.
